### PR TITLE
feat: remove task specific endpoints

### PR DESCRIPTION
**Pull request by sergebouchut2**
Signed-off-by: Serge Bouchut <serge.bouchut@owkin.com>

## Description

:warning: **DO NOT MERGE** until sdk/frontend PRs are merged:
- https://github.com/Substra/substra-frontend/pull/119
- https://github.com/Substra/substra/pull/299
- https://github.com/Substra/substrafl/pull/31
- https://github.com/Substra/substra-tests/pull/221

Remove specific task views.
Update create task views, remove deprecated fields:

Request:
- category
- parent_task_keys
- data_manager_key
- train_data_sample_keys
- test_data_sample_key
- in_model_keys
- predict_model_key

Response:
- category
- parent_task_keys
- train
- test
- aggregate
- composite
- predict

## How has this been tested?

E2e tests [here](https://github.com/owkin/substra-ci/actions/runs/3182568099/jobs/5188675617)

## Checklist

- [x] [changelog](../CHANGELOG.md) was updated with notable changes
- [ ] documentation was updated

### CODE REVIEW

**Comment by sergebouchut2** [comment]
/e2e --refs substra=feat/generic_task,substrafl=feat/generic_task,substra-tests=feat/generic_task,substra-backend=feat/remove_task_specific_endpoints,orchestrator=refactor/remove-task-specific-data-task-category --tests sdk,substrafl

**Comment by Owlfred** [comment]
End to end tests: :hourglass_flowing_sand: **STARTING**

**Comment by sergebouchut2** [comment]
/e2e --refs substra=feat/generic_task,substrafl=feat/generic_task,substra-tests=feat/generic_task,substra-backend=feat/remove_task_specific_endpoints --tests sdk,substrafl

**Comment by Owlfred** [comment]
End to end tests: :hourglass_flowing_sand: **STARTING**

**Comment by Owlfred** [comment]
End to end tests: :hourglass_flowing_sand: **STARTING**

**Comment by sergebouchut2** [comment]
/e2e --refs substra=feat/generic_task,substrafl=feat/generic_task,substra-tests=feat/generic_task,substra-backend=feat/remove_task_specific_endpoints --tests sdk,substrafl

**Comment by sergebouchut2** [comment]
/e2e --refs substra=feat/generic_task,substrafl=feat/generic_task,substra-tests=feat/generic_task,substra-backend=feat/remove_task_specific_endpoints --tests sdk,substrafl

**Comment by Owlfred** [comment]
End to end tests: :hourglass_flowing_sand: **STARTING**

File: CHANGELOG.md [hunk 1]
```diff
@@ -24,10 +24,8 @@ and this project adheres to [Semantic Versioning](https://semver.org/spec/v2.0.0
 ### Removed
 
 - Algo creation events aren't included in newsfeed anymore.
-
-### Changed
-
-- Remove task category from the compute engine.
```
Comment (mblottiere) [inline CHANGELOG.md:30]: Shouldn't this entry be kept?

Comment (sergebouchut2) [reply CHANGELOG.md:30]: sure thx

**Comment by sergebouchut2** [comment]
/e2e --refs substra=feat/generic_task,substrafl=feat/generic_task,substra-tests=feat/generic_task,substra-backend=feat/remove_task_specific_endpoints --tests sdk,substrafl

**Comment by Owlfred** [comment]
End to end tests: :hourglass_flowing_sand: **STARTING**

**Comment by sergebouchut2** [comment]
/e2e --refs substra=feat/generic_task,substrafl=feat/generic_task,substra-tests=feat/generic_task,substra-backend=feat/remove_task_specific_endpoints --tests sdk,substrafl

**Comment by Owlfred** [comment]
End to end tests: :hourglass_flowing_sand: **STARTING**

**Pull request merged**